File: app/channels/post_channel.rb
class PostChannel < ApplicationCable::Channel
  def subscribed
    stream_from "posts_#{current_user.id}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  #def echo(data)
  #  ActionCable.server.broadcast "demo", data
  #end
end

File: app/javascript/channels/post_channel.js
import consumer from "./consumer"

var post = consumer.subscriptions.create("PostChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log(this);
    //this.perform("echo",{
    //  hola: "mundo"
    //})
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log(data)
    if(data.action == 'new_post'){
      $("#posts").prepend(data.message)
    }
    // Called when there's incoming data on the websocket for this channel
  },

  //echo(){
    //@perform "echo", {hola: "mundo"}
  //}
});
